src/Compilers/VisualBasic/Portable/Preprocessor/ExpressionEvaluator.vb

```diff
@@ -852,15 +852,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             If specialType = SpecialType.System_String Then
-                Return ReportSemanticError(ERRID.ERR_CannotConvertValue2, expr)
+                Return ReportSemanticError(ERRID.ERR_UnaryOperand2, expr, expr.OperatorToken.ValueText, specialType.GetDisplayName())
             End If
 
             If specialType = SpecialType.System_Object AndAlso Not IsNothing(val) Then
-                Return ReportSemanticError(ERRID.ERR_CannotConvertValue2, expr)
+                Return ReportSemanticError(ERRID.ERR_UnaryOperand2, expr, expr.OperatorToken.ValueText, specialType.GetDisplayName())
             End If
 
             If specialType = SpecialType.System_Char OrElse specialType = SpecialType.System_DateTime Then
-                Return ReportSemanticError(ERRID.ERR_CannotConvertValue2, expr)
+                Return ReportSemanticError(ERRID.ERR_UnaryOperand2, expr, expr.OperatorToken.ValueText, specialType.GetDisplayName())
             End If
```

Comment:
Consider combining all three cases with ```If A OrElse B OrElse C Then```. (C already has two cases.) Or use Select Case for all cases including `SpecialType.None`.